indexing/src/main/java/com/google/enterprise/cloudsearch/sdk/indexing/IndexingServiceImpl.java

```diff
@@ -794,12 +817,23 @@ public void remove() {
       ListenableFuture<Item> itemUploaded =
           Futures.transform(
               contentUploadService.uploadContent(uploadRef.getName(), content),
-              voidVal ->
-                  item.setContent(
-                      new ItemContent()
-                          .setContentDataRef(uploadRef)
-                          .setHash(contentHash)
-                          .setContentFormat(contentFormat.name())),
+              voidVal -> {
+                if(deleteFileContentAfterUse && content instanceof FileContent) {
+                  try {
+                    ((FileContent) content).getFile().delete();
+                  } catch (Exception e) {
+                    logger.log(
+                            Level.WARNING,
+                            "Error deleting FileContent File: {0}",
```

Comment:
Would a callback (rather than a flag) be better? which could give you even more flexibility in your connector.

In my opinion, deletion of a file should be handled in the customized connector, rather than in SDK. This sounds like fairly rare feature, is quite risky feature to enable in SDK.

Reply:
Yes, Tanmay and I have discussed a callback that could be included as part of the RepositoryDoc. It would give the Connector an opportunity to do cleanup work. But, right now, the Stats system is capturing the onSuccess callback. Is there a way to chain together a second callback (provided as part of the RepositoryDoc?) such that the Stats get updated and then the custom callback is called next?